pkg/alertmanager/api_test.go

```diff
@@ -0,0 +1,109 @@
+package alertmanager
+
+import (
+	"bytes"
+	"context"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/cortexproject/cortex/pkg/alertmanager/alerts"
+	"github.com/cortexproject/cortex/pkg/util"
+
+	"github.com/stretchr/testify/require"
+	"github.com/weaveworks/common/user"
+	"gopkg.in/yaml.v2"
+)
+
+func TestAMConfigValidationAPI(t *testing.T) {
+	testCases := []struct {
+		cfg    UserConfig
+		hasErr bool
+	}{
+		{
+			cfg: UserConfig{
+				AlertmanagerConfig: `
+route:
+  receiver: 'default-receiver'
+  group_wait: 30s
+  group_interval: 5m
+  repeat_interval: 4h
+  group_by: [cluster, alertname]
+`,
+			},
+			hasErr: true,
+		},
+		{
+			cfg: UserConfig{
+				AlertmanagerConfig: `
+route:
+  receiver: 'default-receiver'
+  group_wait: 30s
+  group_interval: 5m
+  repeat_interval: 4h
+  group_by: [cluster, alertname]
+receivers:
+  - name: default-receiver
+`,
+			},
+			hasErr: false,
+		},
+		{
+			cfg: UserConfig{
+				AlertmanagerConfig: `
+route:
+  receiver: 'default-receiver'
+  group_wait: 30s
+  group_interval: 5m
+  repeat_interval: 4h
+  group_by: [cluster, alertname]
+receivers:
+  - name: default-receiver
+`,
+				TemplateFiles: map[string]string{
+					"good.tpl":          "good-template",
+					"not/very/good.tpl": "bad-template", // Paths are not allowed in templates.
+				},
+			},
+			hasErr: true,
+		},
+	}
+
+	am := &MultitenantAlertmanager{
+		store:  noopAlertStore{},
+		logger: util.Logger,
+	}
+	for _, tc := range testCases {
+		payload, err := yaml.Marshal(&tc.cfg)
+		require.NoError(t, err)
+
+		req := httptest.NewRequest("GET", "http://foooo", bytes.NewReader(payload))
+		ctx := context.Background()
+		ctx = user.InjectOrgID(ctx, "testing")
+		require.NoError(t, user.InjectOrgIDIntoHTTPRequest(ctx, req))
+		w := httptest.NewRecorder()
+		am.SetUserConfig(w, req)
+
+		resp := w.Result()
+		if tc.hasErr {
+			require.Equal(t, 400, resp.StatusCode)
+		} else {
```

Comment:
I would check the message as well to ensure the format looks 👍  as this is what users would see.